app/models/user.rb

```diff
@@ -23,6 +27,7 @@ class User < ApplicationRecord
 
   has_many :personal_access_tokens, dependent: :destroy
   has_many :data_exports, dependent: :destroy
+  has_many :namespace_bots, class_name: 'NamespaceBot', dependent: :destroy
```

Comment:
**nitpick (blocking):** Do you want to add a `has_many :bots, through: :namespace_bots ...` so that you can get the bots directly? Or do you just want to pass the `NamespaceBot` to the destroy service potentially so that it and the bot can be deleted?

Reply:
After going through it, it makes sense to pass the namespacebot to the destroy service which will destroy the entity and the underlying user and it's associations (PAT, Member). Updated in adfe4001c792f2935730fcb01ed3c9c93771ea9a. I also added a has_many in this commit to the project_namespace but removed it in the commit after as I don't think we will be needing it (if we do we can always add it back)